src/charm.py

```diff
@@ -355,7 +366,7 @@ def _update_config(self) -> None:
         pending.append((self._amtool_config_path, amtool_config))
 
         # if no config provided, use default config with a dummy receiver
-        config = yaml.safe_load(self.config["config_file"]) or self._default_config
+        config = self._get_config() or self._default_config
```

Comment:
You mean in case of multiple configs we should use remote **and** block at the same time? Why not just block and let the operator decide which config should be used while charm continues working with the current config?

Reply:
Implementation updated.